packages/eyeglass/src/util/SassImplementation.ts

```diff
@@ -0,0 +1,354 @@
+import { render, renderSync, SassError, Options } from "node-sass";
+import { types } from "util";
+import { unreachable } from "./assertions";
+export { Options } from "node-sass";
+
+type SassRenderCallback = Parameters<typeof render>[1];
+type SyncContext<T extends object = {}> = {options: Options & T, callback: undefined};
+type AsyncContext = {options: Options, callback: SassRenderCallback};
+
+export type SyncSassFunction = (this: SyncContext, ...$args: Array<SassValue>) => SassValue | SassError;
+
+export type SassFunctionCallback = ($result: SassValue | SassError) => void;
+export type SassFunction0 = (this: AsyncContext, cb: SassFunctionCallback) => void;
+export type SassFunction1 = (this: AsyncContext, $arg1: SassValue, cb: SassFunctionCallback) => void;
+export type SassFunction2 = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, cb: SassFunctionCallback) => void;
+export type SassFunction3 = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, cb: SassFunctionCallback) => void;
+export type SassFunction4 = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue, cb: SassFunctionCallback) => void;
+export type SassFunction5 = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue, $arg5: SassValue, cb: SassFunctionCallback) => void;
+export type SassFunction6 = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue, $arg5: SassValue, $arg6: SassValue, cb: SassFunctionCallback) => void;
+
+export type SassFunction1Var = (this: AsyncContext, $arg1: SassValue[], cb: SassFunctionCallback) => void;
+export type SassFunction2Var = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue[], cb: SassFunctionCallback) => void;
+export type SassFunction3Var = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue[], cb: SassFunctionCallback) => void;
+export type SassFunction4Var = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue[], cb: SassFunctionCallback) => void;
+export type SassFunction5Var = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue, $arg5: SassValue[], cb: SassFunctionCallback) => void;
+export type SassFunction6Var = (this: AsyncContext, $arg1: SassValue, $arg2: SassValue, $arg3: SassValue, $arg4: SassValue, $arg5: SassValue, $arg6: SassValue[], cb: SassFunctionCallback) => void;
+
+export type SassFunction =
+  SyncSassFunction
+  | SassFunction0 | SassFunction1 | SassFunction2 | SassFunction3 | SassFunction4 | SassFunction5 | SassFunction6
+  | SassFunction1Var | SassFunction2Var | SassFunction3Var | SassFunction4Var | SassFunction5Var | SassFunction6Var;
+
+export interface FunctionDeclarations {
+  [functionDeclaration: string]: SassFunction;
+}
```

Comment:
This could also be a utility type
```ts
type FunctionDeclarations = Record<string, SassFunction>;
```

Reply:
Fixed in 47f8e3e2d1be344cd36949475f40df01bb5c1b69.